compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2956,6 +2958,133 @@ pub(crate) fn are_upstream_rust_objects_already_included(sess: &Session) -> bool
     }
 }
 
+/// We need to communicate four things to the linker on Apple/Darwin targets:
+/// - The architecture.
+/// - The operating system (and that it's an Apple platform).
+/// - The deployment target.
+/// - The environment / ABI.
+fn add_apple_link_args(cmd: &mut dyn Linker, sess: &Session, flavor: LinkerFlavor) {
+    if !sess.target.is_like_osx {
```

Comment:
Can use `let LinkerFlavor::Darwin(cc, _) = flavor else { return; };` here instead to avoid the FIXME below.
They mean the same thing, we report an error if linker flavor is incompatible with the current target.